services/src/kilda-core/kilda-persistence-api/src/main/java/org/openkilda/persistence/repositories/SwitchRepository.java

```diff
@@ -26,4 +28,8 @@
     Optional<Switch> findById(SwitchId switchId);
 
     Switch reload(Switch entity);
+
+    Iterable<FlowSegment> findFlowSegmentsToSwitch(SwitchId switchId);
```

Comment:
`findIncomingSomeRelation` is more ... natural. Also in this case we can avoid word `switch` in this case.